src/platform/site-wide/user-nav/selectors.js

```diff
@@ -0,0 +1,18 @@
+import { createSelector } from 'reselect';
+import _ from 'lodash';
+import { selectProfile } from '../../user/selectors';
+
+export const selectUserGreeting = createSelector(
```

Comment:
I pulled this out because it lets us calculate the props we need instead of passing around the whole `profile` state slice.